includes/Modules/Analytics_4.php

```diff
@@ -1469,6 +1491,44 @@ protected function create_data_request( Data_Request $data ) {
 						self::normalize_property_id( $data['propertyID'] ),
 						$custom_dimension
 					);
+
+			case 'GET:audience-settings':
+				return function () {
+					$settings = $this->audience_settings->get();
+					return current_user_can( Permissions::MANAGE_OPTIONS ) ? $settings : array_intersect_key( $settings, array_flip( $this->audience_settings->get_view_only_keys() ) );
+				};
+
+			case 'POST:save-audience-settings':
+				if ( ! current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+					return new WP_Error(
+						'forbidden',
+						__( 'User does not have permission to save audience settings.', 'google-site-kit' ),
+						array( 'status' => 403 )
+					);
+				}
+
+				if ( isset( $data['audienceSegmentationSetupCompletedBy'] ) && ! is_int( $data['audienceSegmentationSetupCompletedBy'] ) ) {
+					throw new Invalid_Param_Exception( 'audienceSegmentationSetupCompletedBy' );
+				}
+
+				return function () use ( $data ) {
+					if ( isset( $data['audienceSegmentationSetupCompletedBy'] ) ) {
+						$new_settings['audienceSegmentationSetupCompletedBy'] = $data['audienceSegmentationSetupCompletedBy'];
+					}
+
+					if ( isset( $data['availableAudiences'] ) && is_array( $data['availableAudiences'] ) ) {
+						$new_settings['availableAudiences'] = $data['availableAudiences'];
+					}
+
+					if ( isset( $data['availableAudiencesLastSyncedAt'] ) ) {
+						$new_settings['availableAudiencesLastSyncedAt'] = intval( $data['availableAudiencesLastSyncedAt'] );
+					}
+
```

Comment:
As per my previous comment, we shouldn't handle settings other than `audienceSegmentationSetupCompletedBy` in the `POST:save-audience-settings` datapoint.

> we don't want to set `audienceSegmentationSetupCompletedBy` to the current user ID here, rather the idea is we should allow the setting to be passed in, but not handle any other settings for now

I realise my suggested edit didn't make this entirely clear. I'll apply this change myself to save the round trip.

```suggestion
```